c++/src/capnp/serialize-async.h

```diff
@@ -76,6 +76,107 @@ kj::Promise<void> writeMessage(kj::AsyncCapabilityStream& output, kj::ArrayPtr<c
     KJ_WARN_UNUSED_RESULT;
 // Write a message with FDs attached, e.g. to a Unix socket with SCM_RIGHTS.
 
+// -----------------------------------------------------------------------------
+// Versions virtualized over an interface.
+
+class MessageTransport {
```

Comment:
`MessageStream` might be a clearer name. "Transport" is more abstract and I sometimes use it to describe a network rather than an individual stream.

I wonder if it makes any sense to have separate `MessageInputStream` and `MessageOutputStream`. But I guess it doesn't matter for the immediate use case, and it's possible to refactor later if we want.